### PR TITLE
Fix Interval#overlaps()

### PR DESCRIPTION
**Pull request by kevincox**
Upstream PR: https://github.com/intiocean/pyinter/pull/16

@angelini @datwright @solackerman @kmtaylor-github

### CODE REVIEW

File: pyinter/interval.py [hunk 1]
```diff
@@ -203,10 +203,13 @@ def empty(self):
 
     def overlaps(self, other):
         """If self and other have any overlaping values returns True, otherwise returns False"""
-        if self.lower_value in other or self._upper_value in other or \
-                other.lower_value in self or other._upper_value in self or self == other:
-            return True
-        return False
+        if self > other:
+            self, other = other, self
```
Comment (kmtaylor-github) [inline pyinter/interval.py:207]: As a Smalltalk / Java guy, this just makes me ill  :smile:

Comment (kevincox) [inline pyinter/interval.py:207]: Yeah, originally I was calling `return other.overlaps(self)` but then I realized that that was completely unnecessary :P

**Comment by kmtaylor-github** [comment]
A bit tough to follow, but logic is sound. 
The empty corner cases are interesting (since you can have a large or small low_value and still be empty), and probably worth testing.

**Comment by kevincox** [comment]
Ok, I added some test cases for empty intervals (both high and low).

**Comment by kmtaylor-github** [comment]
LGTM
